### PR TITLE
Hooked up and tested --output and --clobber options

### PR DESCRIPTION
**Pull request by noahmacaskill**
Had a go at some tests for --output, also created the --clobber option you suggested. Let me know what you think, especially on the tests. I tried to think of the simplest way to do them, opted to use version to test the output. 

Also tested that it fails when attempting an overwrite without using the --clobber option, then passes when the option is used. I experimented with asserting the correct error message is being displayed when --clobber isn't used but couldn't quite get it to work properly. Can have another go at that if simply asserting a return value of 1 isn't enough.

### CODE REVIEW

File: test/MetadataUtility.Tests/Cli/OutputSinkTests.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            FixtureHelper.TestTempDir tempDir = new FixtureHelper.TestTempDir();
+            string path = Path.Join(tempDir.TempDir, "out.txt");
+
+            var result = await EmuEntry.Main(new string[] { "version", "-O", path });
+
+            Assert.Equal(result, 0);
+
+            await this.command.InvokeAsync(null);
+
+            Assert.Equal(sw.ToString(), File.ReadAllText(path));
```
Comment (atruskie) [inline test/MetadataUtility.Tests/Cli/OutputSinkTests.cs:45]: oh i understand now. Add some comments maybe to describe what you're thinking here?

I'd suggest another simpler test to add as well:

Capture standard output, run the command, assert standard output is empty, and finally assert the output file has the expected number of lines (maybe match content as well).

File: test/MetadataUtility.Tests/Cli/OutputSinkTests.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+            tempDir.Dispose();
```
Comment (atruskie) [inline test/MetadataUtility.Tests/Cli/OutputSinkTests.cs:47]:
```suggestion
```

File: test/MetadataUtility.Tests/Cli/OutputSinkTests.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        [Fact]
+        public async void testOutputOverwriting()
+        {
+
+            FixtureHelper.TestTempDir tempDir = new FixtureHelper.TestTempDir();
+            string path = Path.Join(tempDir.TempDir, "out.txt");
+
+            File.WriteAllLines(path, new string[] { "test" });
```
Comment (atruskie) [inline test/MetadataUtility.Tests/Cli/OutputSinkTests.cs:57]:
```suggestion
            File.WriteAllTextpath, "test");
```

File: test/MetadataUtility.Tests/Cli/OutputSinkTests.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+            var result = await EmuEntry.Main(new string[] { "version", "-O", path });
+
+            Assert.Equal(result, 1);
+
```
Comment (atruskie) [inline test/MetadataUtility.Tests/Cli/OutputSinkTests.cs:62]: add an assertion for the output includes an error message that explains to the user what is wrong

**Review by atruskie — CHANGES_REQUESTED** [review]
Looks like a great set of changes.

I'm concerned with the UX of the clobber exception. Does it print a stack trace? What does it look like in the console?

General points:
- PascalCase for naming methods please
- Generally we avoid using full namespaces for types unless they absolutely can't be resolved


Sorry for the late review, I thought I submitted this yesterday

**Comment by noahmacaskill** [comment]
Made most of the small changes, running into a few small issues with a couple things.

As for the clobber exception, yes it's a pretty ugly stack trace. Was trying to think of some possible solutions here. Maybe we set up a flag in OutputRecordWriter to deal with it in there? Or we set the output sink to null and deal with the null pointer exception where appropriate (thinking not because the null pointer exception may not have to do with the clobber option). Let me know if you have any suggestions.

**Comment by noahmacaskill** [comment]
Replaced the ugly stack trace with a nice error log message as discussed, but the tests may still need some work. 

In the tests I'm struggling to grab the error message from the log, since I'm running this directly from the EmuEntry main method, I'm not sure how to rewire the output to a string writer like I did before (since before I could do so via the command constructor). 

Let me know your suggestions, I'm thinking I might shift my focus to metadata extraction next week and circle back to this later.

**Comment by atruskie** [comment]
@noahmacaskill I wanted a clean base to work off of for the metadata extractor work. So I went ahead and finished this branch off - I wan't quite sure how to achieve it so just playing with it myself was more effective I thought than guiding you through several more attempts. 

Thanks for the head start though!

**Pull request closed without merging**